src-plugins/mscExportVideo/mscExportVideo.cpp

```diff
@@ -0,0 +1,283 @@
+#include "mscExportVideo.h"
+
+#include <dtkCore/dtkAbstractProcessFactory.h>
+
+#include <vtkImageCanvasSource2D.h>
+#include <vtkImageData.h>
+#include <vtkJPEGWriter.h>
+#include <vtkOggTheoraWriter.h>
+#include <vtkQImageToImageSource.h>
+#include <vtkSmartPointer.h>
+
+// /////////////////////////////////////////////////////////////////
+// ExportVideoPrivate
+// /////////////////////////////////////////////////////////////////
+namespace msc
+{
+class ExportVideoPrivate
+{
+public:
+    QVector< QImage > imagesArray;
+    int width;
+    int height;
+
+    // User parameters
+    QString filename;
+    int format;            // Cf. VideoFormat in header file
+    int frameRate;         // Frame per second
+    bool subsampling;      // Is the video to be encoded using 4:2:0 subsampling?
+    int quality;           // 0 = low, 1 = medium, 2 = high
+    bool exportOnlyAsJPEG; // Can be used for home encoding
+};
+
+// /////////////////////////////////////////////////////////////////
+// ExportVideo
+// /////////////////////////////////////////////////////////////////
+
+ExportVideo::ExportVideo() : medAbstractProcess(), d(new ExportVideoPrivate)
+{
+    d->imagesArray.clear();
+    d->width  = 0;
+    d->height = 0;
+
+    // User parameters
+    d->format = OGGVORBIS;
+    d->filename = "";
+    d->frameRate = 1;
+    d->subsampling = false;
+    d->quality = 2;
+    d->exportOnlyAsJPEG = false;
+}
+
+ExportVideo::~ExportVideo()
+{
+    delete d;
+}
+
+bool ExportVideo::registered()
+{
+    return dtkAbstractProcessFactory::instance()->registerProcessType("msc::ExportVideo", createExportVideo);
+}
+
+QString ExportVideo::description() const
+{
+    return "ExportVideo";
+}
+
+QString ExportVideo::identifier() const
+{
+    return description();
+}
+
+void ExportVideo::setParameter(int* data)
+{
+    d->width  = data[0];
+    d->height = data[1];
+
+    QImage currentImage = QPixmap(d->width, d->height).toImage();
+
+    for (int i=0; i<d->width; ++i)
+    {
+        for (int j=0; j<d->height; ++j)
+        {
+            // R, G, B to QColor (A default at 255)
+            int index1D = 2 + (i * d->height + j)*3;
+            QColor color = QColor(data[index1D    ],
+                                  data[index1D + 1],
+                                  data[index1D + 2]);
+
+            currentImage.setPixel(i, j, color.rgb());
+        }
+    }
+
+    d->imagesArray.append(currentImage);
```

Comment:
No need to use insert, it's not the tool design. We record "on the fly" screenshots of the view.

Reply:
(just updated the code I posted, there was a minor error)